src/pe/mod.rs

```diff
@@ -68,6 +69,21 @@ impl<'a> PE<'a> {
             + header::SIZEOF_PE_MAGIC
             + header::SIZEOF_COFF_HEADER
             + header.coff_header.size_of_optional_header as usize);
+
+        let min_size = *offset
+            + header.coff_header.number_of_sections as usize * header::SIZEOF_IMAGE_SECTION_HEADER;
+
+        if bytes.len() < min_size {
+            // The section table contains more entries than the binary can provide given the size
+            return Err(error::Error::Malformed(format!(
+                "Corrupted PE: Expected at least {:#X} bytes but got {:#X}",
+                min_size,
+                bytes.len()
+            )));
+        }
+
+        let section_table = &bytes[*offset..min_size];
```

Comment:
~~shouldn't this be `&bytes[*offset..*offset + min_size]` ?~~

Nit: took me a second to realize `min_size` includes the offset; a better name might be `section_end_offset` or something? I guess `min_size` makes more sense when it's compared to `bytes.len()`, but usually when i see `offset..size` i expect a `offset..offset+size` or `start..end` where start = offset, and end is offset + sizeofthings.

Anyway, just me rambling a bit haha

Reply:
I agree, `min_size` is a bit too generic. I've changed it to `section_end_offset`.